Source/Core/Core/HW/DVDThread.cpp

```diff
@@ -47,9 +47,9 @@ static u32 s_output_address;
 static u32 s_length;
 static bool s_decrypt;
 
-// Used to notify emulated software after executing command.
-// Pointers don't work with savestates, so CoreTiming events are used instead
-static int s_callback_event_type;
+// This determines which function will be used as a callback.
+// We can't have a function pointer here, because they can't be in savestates.
+static bool s_reply_to_ios;
```

Comment:
Indeed. That's what I was planning to do, so it provides a transparent interface (and is less intrusive to everything else). Something like the following should suffice (note: I have not tested this, this is entirely conjecture).

```cpp
template <>
void Do(bool& value)
{
    // Make the given bool value a 32-bit value to be stable across all platforms
    // where bool may differ in terms of bit size.
    u32 stable = static_cast<u32>(value);

    DoVoid(&stable, sizeof(u32));

    if (mode == MODE_READ)
         value = stable != 0;
}
```

would you prefer a change like this to be included in 5.0, or after?